src/components/ExceededCommentLength.js

```diff
@@ -14,47 +14,27 @@ const propTypes = {
     onExceededMaxCommentLength: PropTypes.func.isRequired,
 };
 
-class ExceededCommentLength extends PureComponent {
-    constructor(props) {
-        super(props);
-
-        this.state = {
-            commentLength: 0,
-        };
-
-        // By debouncing, we defer the calculation until there is a break in typing
-        this.updateCommentLength = debounce(this.updateCommentLength.bind(this), CONST.TIMING.COMMENT_LENGTH_DEBOUNCE_TIME);
-    }
-
-    componentDidMount() {
-        this.updateCommentLength();
+function ExceededCommentLength(props) {
+    const [commentLength, setCommentLength] = useState(0);
+    const updateCommentLength = useMemo(
+        () =>
+            debounce((comment, onExceededMaxCommentLength) => {
+                const newCommentLength = ReportUtils.getCommentLength(comment);
+                setCommentLength(newCommentLength);
+                onExceededMaxCommentLength(newCommentLength > CONST.MAX_COMMENT_LENGTH);
+            }, CONST.TIMING.COMMENT_LENGTH_DEBOUNCE_TIME),
+        [],
+    );
+
+    useEffect(() => {
+        updateCommentLength(props.comment, props.onExceededMaxCommentLength);
+    }, [props.comment, props.onExceededMaxCommentLength, updateCommentLength]);
```

Comment:
Without adding the `updateCommentLength` to the dependency array, the eslint is throwing the `react-hooks/exhaustive-deps` warning. It is a bit annoying, as it will never change.

I could add a `eslint-disable-next-line` with an additional comment, but the current solution seems simpler and less error-prone in the future.